core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1038,9 +1038,8 @@ class DAGScheduler(
     }
 
     if (tasks.size > 0) {
-      logInfo("Submitting " + tasks.size + " missing tasks from " + stage + " (" + stage.rdd + ")")
-      stage.pendingPartitions ++= tasks.map(_.partitionId)
-      logDebug("New pending partitions: " + stage.pendingPartitions)
+      logInfo(s"Submitting ${tasks.size} missing tasks (for partitions " +
+        s"${tasks.map(_.partitionId)}) from $stage (${stage.rdd})")
```

Comment:
Oh good point.  25 seemed like maybe too many for an info-level message so I split the difference with your original comment and did 15.  I could also just leave the original logging if you think that's better.